### DISCUSSION TITLE
Trying to understand the logic(or magic) why this is working too perfectly?

### DISCUSSION THREAD

**Discussion by cybervaldez** [statelyai/xstate#1410] [General] [answered]
Here's how my machine is laid out
```
#formMachine
- context: { isGroupFocused : false }
-- names (state)
--- firstName (spawned, context : { isFocused : false })
--- middleName (spawned, context : { isFocused : false })
--- lastName (spawned, context : { isFocused : false })

```
The spawned machines are all tied to their input elements and when blurred/focused sends [assign({ isFocused : true/false }), sendUpdate()] respectively.

I've had this setup so when one of the fields are in focus, the machine's context will be { isGroupFocused : true } which would have my ui component highlighting the fields parent container. What's odd is this working too perfectly for me, in a sense that when tabbing to change inputs I would not see the focus/unfocus state. Before xstate I did a similar thing which required me to do some debouncing just to prevent those flashes so i'm just wondering why this isn't the case for xstate.

Here's what I was expecting:
1. Focus on an input (Group gets highlighted)
2. Pressing tab to move to the next field would trigger first a BLUR and then a FOCUS (which very briefly shows the group highlighting out and in again)

It's great that it's working this way btw (as intended) but I'd just like to understand why this is the case.

**Comment by Andarist** [comment, marked as answer]
You'd really have to share your current and previous solutions so we could compare how they differ in behavior. It's very hard to answer your question just based on the given description.

**Reply by cybervaldez** [reply]
Sorry about that. Just realized my last approach I was using useEffect which was causing the issue.

https://codesandbox.io/s/priceless-murdock-59k2i?file=/src/index.js
